sky/spot/spot_utils.py

```diff
@@ -507,3 +507,37 @@ def load_job_table_cache() -> Tuple[str, str]:
         return None
     with cache_file.open('r') as f:
         return json.load(f)
+
+
+def is_spot_controller_up(
+    stopped_message: str,
+) -> Tuple[Optional[global_user_state.ClusterStatus],
+           Optional[backends.Backend.ResourceHandle]]:
+    try:
+        controller_status, handle = backend_utils.refresh_cluster_status_handle(
+            SPOT_CONTROLLER_NAME, force_refresh=True)
+    except (exceptions.CloudUserIdentityError,
+            exceptions.ClusterOwnerIdentityMismatchError,
+            exceptions.ClusterStatusFetchingError) as e:
+        logger.warning(
+            f'Failed to get the status of the spot controller. '
+            'It is not fatal, but spot commands may hang, when '
+            'the controller is not up, or have inaccurate statement.\n'
```

Comment:
For 1, the inaccurate statement means the caller can print out "all the spot jobs are finished" or claim the spot controller in some status when it is not.

For 2, it can hang for about 10 seconds when trying to ssh into a stopped controller.